File: springboot-web/json-web-examples/src/main/java/techtabu/json/mapkey/PersonKeyWithJ.java
package techtabu.json.mapkey;

import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Data;

/**
 * @author TechTabu
 */

@Data
@AllArgsConstructor
public class PersonKeyWithJ {

    private String firstName;
    private String lastName;

    @JsonValue
    public String convertToKey() {
        return "{firstName: " + firstName + ", lastName: " + lastName + "}";
    }
}

File: springboot-web/json-web-examples/src/main/java/techtabu/json/mapkey/PersonKeyWithJ.kt
package techtabu.json.mapkey

import com.fasterxml.jackson.annotation.JsonValue
import lombok.AllArgsConstructor
import lombok.Data

/**
 * @author TechTabu
 */
@Data
@AllArgsConstructor
class PersonKeyWithJ(firstName: String, lastName: String) {
    private val firstName: String? = firstName
    private val lastName: String? = lastName
    @JsonValue
    fun convertToKey(): String {
        return "{firstName: $firstName, lastName: $lastName}"
    }
}